x-pack/plugins/observability_solution/dataset_quality/common/plugin_config.ts

```diff
@@ -5,5 +5,23 @@
  * 2.0.
  */
 
-// eslint-disable-next-line @typescript-eslint/no-empty-interface
-export interface DatasetQualityConfig {}
+import { offeringBasedSchema, schema, TypeOf } from '@kbn/config-schema';
+import { PluginConfigDescriptor } from '@kbn/core-plugins-server';
+
+const configSchema = schema.object({
+  estimatedDataEnabled: offeringBasedSchema({
+    traditional: schema.boolean({ defaultValue: true }),
```

Comment:
This would allow setting it on traditional instances, is this something we want to allowed?

Reference: https://github.com/elastic/kibana/blob/e98b4b3e88a1a5774a3c96429adbccc109eb4e34/packages/kbn-config-schema/src/helpers/offering_based_schema.ts#L25

For how to specify serverless-only settings that are not exposed on traditional.

If there is a good reason to allow on-prem users to disable this via config, we should leave it, otherwise we should hide it for traditional so it's not a breaking change once we remove it.

Reply:
We could also rely on `coreStart.elasticsearch.getCapabilities().serverless` as this difference in behavior is ultimately driven by Elasticsearch capabilities

Reply:
We don't need it configurable for on-prem and if `coreStart.elasticsearch.getCapabilities().serverless` serves the purpose, we don't need it configurable at all IMO. Also considering the two points mentioned [in](https://github.com/elastic/kibana/pull/164850).

Reply:
It's definitely a nit as it doesn't change much right now, but this seems the right way "in spirit", which means it will cause the least amount of problems in the future.

Reply:
@flash1293 👍👍 I've switched this to an Elasticsearch capabilities / route driven approach (since we wouldn't have this information available via the client).